site-src/geps/gep-1709.md

```diff
@@ -37,8 +45,296 @@ which implementations can prove they satisfy and be recognized for.
+the corresponding conformance profile.
+
+Unlike `Core` the `Extended` support level presents a slight problem with the
+underlying basis on `SupportedFeatures` as implementations might support some
+(but not all) of a the `Extended` feature set. For implementations that intend
+to support _partial_ `Extended` features they will be required to _exclude_ any
+of the `Extended` features they don't support with test arguments (which we
+will cover in the [integration section](#integration)) and these exemptions
+will be reflected during the [reporting process](#reporting-process).
+
+> **NOTE**: There are situations where an implementation might actually be able
+> to fulfill conformance tests for multiple profiles (e.g. `Layer4Core` and
+> `Layer7Extended`, or perhaps an implementation can literally do everything,
+> including `Mesh`!). Rather than bother with even higher level compilations at
+> this point, we will simply report these separately and the implementations
+> can indicate that they support multiples to their end-users in whatever way
+> best suits them. If over time we find that the community is really strongly
+> in favor of compilations at a higher level we are set up to add them in a
+> follow up iteration.
+
+[camelcase]:https://en.wikipedia.org/wiki/Camel_case
+[feat]:https://github.com/kubernetes-sigs/gateway-api/blob/c61097edaa3b1fad29721e787fee4b02c35e3103/conformance/utils/suite/suite.go#L33
+
+### Integration
+
+Integrating the test suite into your implementation can be done using one of
+the following methods:
+
+- The [go test][go-test] command line interface which enables projects of any
+  language to run the test suite on any platform [Golang][go] supports.
+- Using the conformance test suite as a [Golang library][lib] within an already
+  existing test suite.
+
+> **NOTE**: Usage as a library is already an established colloquialism in the
+> community, this effort simply intends to make that more official.
+
+Selected conformance profiles can be identified as arguments in either
+configuration. For instance using the command line interface method to run the
+`Layer7Core` profile to verify core feature conformance:
+
+```console
+$ go test ./conformance/... -args -gateway-class=acme -conformance-profile=Layer7Core
+```
+
+Or the equivalent configuration using the Golang library:
+
+```go
+cSuite, err := suite.New(suite.Options{
+    GatewayClassName: "acme",
+    Profile: Layer7CoreProfile,
+    // other options
+})
+require.NoError(t, err, "misconfigured conformance test suite")
+cSuite.Setup(t)
+
+for i := 0; i < len(tests.ConformanceTests); i++ {
+    test := tests.ConformanceTests[i]
+    test.Run(t, cSuite)
+}
+```
+
+> **NOTE**: In the `suite.Options` above it's still possible to add `SkipTests`
+> but when used in conjunction with `Profile` this will result in a report that
+> the profile is not valid for reporting. Implementations in this state may be
```

Comment:
why not add a section in the report highlighting what was `Skipped` ?

Reply:
My intention here was to _never_ report conformance when tests are explicitly skipped, instead making `SkipTests` a tool for the development phase. Can you help me to understand the value you see in allowing reporting where entire tests are skipped, even if they might have been required? In the case of extended where we know some _features_ may not be fully supported, wouldn't `UnsupportedFeatures` adequately cover things here?

Reply:
`SkipTests` are fewer, more fine grained tests, within `SupportedFeatures`, that the project cannot support either due to a bug in the project or due to API clarification. Here the setting in [Envoy Gateway](https://github.com/envoyproxy/gateway/blob/afe646b84d03cf4126f55721589850dc202dc130/test/conformance/conformance_test.go#L50). 

With above decision, Envoy Gateway would continue to run all tests, except some with SkipTests (temporarily until issues are fixed), but would not be able to generate any report

Reply:
Right, I think we're kinda saying the same thing. But where I was kinda coming from, is that if you're _skipping tests_ then this basically means you're **not** conformant. I'm a little confused about having a reporting option for implementations which are not yet conformant, but I guess I could be persuaded? :thinking:

Reply:
Sure from a reporting standpoint, you can mark it as `not` conformant :)
From an implementation perspective, we prefer this approach to ensure more test coverage and minimal regression
TLDR, would prefer if the reporting tool marks the specific features (whose tests are skipped) as non conformant rather than fail reporting for other features too

Reply:
The conformance tests run in CI in EG to prevent regression as well as report conformance (e.g. X feature tests pass). Earlier I was thinking we could this plug into the CI and run the tests once , so the report is generated on every PR run, so EG would replace this suite flag `EnableAllSupportedFeatures: true,` with `Profile: Layer4CoreProfile,Layer4ExtendedProfile,Layer7CoreProfile,Layer7ExtendedProfile`. 

But thinking more on this, the conformance profile tests might need to run as a separate task who's job is to only report conformance rather than use the exit code of the process to decide whether CI is passing or not (catch regression).

Reply:
So it just occurred to me a couple days ago, I had already added a note about being able to explicitly report "in progress" conformance via this process:

https://github.com/kubernetes-sigs/gateway-api/pull/1754/files#diff-7764ccb47f68a80348b14a2c6df6c6cadd80db02de7817dbcb8039a611bf1328R130

Also I've removed the support levels on the profiles in favor of a nested reporting structure. If I'm understanding what you were asking for, I think this will provide the facility you were looking for. If not, please mark this unresolved and let me know!